catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/TransformOperations.java

```diff
@@ -43,43 +44,37 @@ public TransformOperations(FrameworkProperties frameworkProperties) {
   //
   // Delegate methods
   //
-  public BinaryContent transform(
+  public List<BinaryContent> transform(
       Metacard metacard, String transformerId, Map<String, Serializable> requestProperties)
       throws CatalogTransformerException {
     if (metacard == null) {
       throw new IllegalArgumentException("Metacard is null.");
     }
 
-    ServiceReference[] refs;
-    try {
-      // TODO replace shortname with id
-      refs =
-          frameworkProperties
-              .getBundleContext()
-              .getServiceReferences(
-                  MetacardTransformer.class.getName(),
-                  "(|"
-                      + "("
-                      + Constants.SERVICE_SHORTNAME
-                      + "="
-                      + transformerId
-                      + ")"
-                      + "("
-                      + Constants.SERVICE_ID
-                      + "="
-                      + transformerId
-                      + ")"
-                      + ")");
-    } catch (InvalidSyntaxException e) {
-      throw new IllegalArgumentException("Invalid transformer shortName: " + transformerId, e);
-    }
-    if (refs == null || refs.length == 0) {
+    List<MultiMetacardTransformer> multiMetacardTransformers =
+        frameworkProperties
+            .getTransformerLocator()
+            .findMultiMetacardTransformersByFilter(
```

Comment:
❗️ Your find by ID method should be doing the same logic.  SERVICE_SHORTNAME is the old way of registering a transformer and SERVICE_ID is the new way.  You should be able to remove the find by filter after doing this.